variation-commons-core/src/test/java/uk/ac/ebi/eva/commons/core/models/AbstractVariantTest.java

```diff
@@ -0,0 +1,122 @@
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        variant.renormalize(-1, 1, "A", "T");
+    }
+
+    @Test
+    public void testNegativeEndCoordinateInConstructor() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant end coordinate cannot be negative"));
+        new Variant("1", 1, -1, "C", "T");
+    }
+
+    @Test
+    public void testNegativeEndCoordinateInRenormalization() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant end coordinate cannot be negative"));
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        variant.renormalize(1, -1, "A", "T");
+    }
+
+    @Test
+    public void testStartHigherThanEndInConstructor() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(
+                CoreMatchers.equalTo("End position must be equal or greater than the start position"));
+        new Variant("1", 2, 1, "C", "T");
+    }
+
+    @Test
+    public void testStartHigherThanEndInRenormalization() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(
+                CoreMatchers.equalTo("End position must be equal or greater than the start position"));
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        variant.renormalize(1, 0, "A", "T");
+    }
+
+    @Test
+    public void testStartEndAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        assertEquals(1, variant.getStart());
+        assertEquals(1, variant.getEnd());
+    }
+
+    @Test
+    public void testStartEndAssignmentInRenormalization() {
+        Variant variant = new Variant("1", 2, 2, "C", "T");
+        variant.renormalize(1, 2, "AC", "AT");
+        assertEquals(1, variant.getStart());
+        assertEquals(2, variant.getEnd());
+    }
+
+    @Test
+    public void testRefAltAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        assertEquals("C", variant.getReference());
+        assertEquals("T", variant.getAlternate());
+
+        variant = new Variant("1", 1, 1, "", "T");
+        assertEquals("", variant.getReference());
+        assertEquals("T", variant.getAlternate());
+
+        variant = new Variant("1", 1, 1, "T", "");
+        assertEquals("T", variant.getReference());
+        assertEquals("", variant.getAlternate());
+    }
+
+    @Test
+    public void testRefAltAssignmentInRenormalization() {
+        Variant variant = new Variant("1", 1, 1, "", "T");
+        variant.renormalize(1, 2, "A", "AT");
+        assertEquals("A", variant.getReference());
+        assertEquals("AT", variant.getAlternate());
+
+        variant = new Variant("1", 1, 1, "T", "");

```

Comment:
I think it is enough for the test to assign and check only once.

Reply:
These tests are to exercise the code path for reference and alternate being empty.